### PR TITLE
updated pyproject.toml with optional flavors on install and in test g…

### LINKED ISSUE #105: Make PyTorch and TensorFlow flavor support testable in CI
Follow-up to #73. We have new flavors that we support but we can't yet test our support for them in CI. Also, it would be good for new developers to optionally install the backing libraries in the same way that they can optionally install the libraries used for testing.

# Assumptions:
1. PyTorch and TensorFlow should not be included in the default garden-ai PyPI package.
2. There should be a way for devs to issue a poetry command that installs versions of PyTorch and TensorFlow that are known to work/install well with the rest of the libraries we depend on.

# Acceptance Criteria
Given a new PR after this PR, when we run CI/CD, then there will be tests run that exercise our flavor support. But pytorch and tensorflow won't be included in our default base garden-ai package.

### PR DESCRIPTION
**Pull request by steve-barnard**
…roup and added tensorflow test, all ml flavors passing in local tests

`Fixes #105`
(https://github.com/Garden-AI/garden/issues/105) 

## Overview

Provides an update to the poetry pyproject.toml to allow for developers/users to optionally install PyTorch or TensorFlow via the following commands:

PyTorch:
`poetry install --extras "torch"`
TensorFlow:
`poetry install --extras "tensorflow"`
all flavors:
`poetry install --all-extras`

These are also included in the optional test group dependencies which should allow for CI testing via the following code invoked during the git action workflow:

testing in general:
`poetry install --with=test`

git action workflow commands:
`poetry install --no-root --no-interaction --no-ansi --with=test`
`python -m coverage run -m pytest -m "not integration" -r sx`

## Discussion

I also updated the tests>test_mlflow.py by adding in a fixture and test for a toy TensorFlow model and was able to get all three supported flavors to pass.

## Testing

I tested out all of the above code to ensure that as the different arguments were provided following poetry installs that proper packages were installed and or removed according to no options, all extras, all flavors, each individually, and with the optional test. I then ran the tests to ensure that the same level of passing tests matched that of the current main branch (other than the one new passing test for TensorFlow upload).

----
:books: Documentation preview :books:: https://garden-ai--107.org.readthedocs.build/en/107/

### CODE REVIEW

**Review by WillEngler — COMMENTED** [review]
Aside from one cleanup nit this looks perfect. If those are in by mistake can you take them out? Then we're good to pull this in.

File: tests/integrations/test_mlflow.py [hunk 1]
```diff
@@ -94,3 +129,28 @@ def test_mlflow_pytorch_register(tmp_path, toy_pytorch_model):
...
+@pytest.mark.integration
+def test_mlflow_tensorflow_register(tmp_path, toy_tensorflow_model):
+    # as if model.pkl already existed on disk
+    # import tensorflow as tf  # type: ignore
+
+    tmp_path.mkdir(exist_ok=True)
+    model_path = tmp_path / "tensorflowtest"
+    toy_tensorflow_model.save(model_path)
+    flavor = "tensorflow"
+    # model_path.parent.mkdir(exist_ok=True)
```
[the remark this answers is not in the document]
Comment (steve-barnard) [reply tests/integrations/test_mlflow.py:143]: Those are likely leftover from my merge to my fork; I can take them out and commit, then we should be all set.

[the remark this answers is not in the document]
Comment (steve-barnard) [reply tests/integrations/test_mlflow.py]: @WillEngler comments should now be removed/updated

**Comment by WillEngler** [comment]
Could you resolve the conflicts? Then we'll be good to pull this in?

**Comment by steve-barnard** [comment]
@WillEngler merge conflicts should be resolved 🤞🏻, may have to update the poetry lock again to confirm, looks to be working locally; the only unknown from me was the content-hash; which should update upon running `poetry lock --no-update` if I recall correctly.

**Comment by steve-barnard** [comment]
@WillEngler I think this should be ok, but unsure about the codecov issues causing the builds to fail.

File: pyproject.toml [hunk 2]
```diff
@@ -15,6 +15,7 @@ pydantic = "^1.10.2"
 typer = {extras = ["all"], version = "^0.7.0"}
 beartype = "^0.12.0"
 jinja2 = "^3.1.2"
+funcx = "^1.0.11"
```
Comment (WillEngler) [inline pyproject.toml:18]: I think this is a merge artifact that got left in by mistake

Comment (steve-barnard) [reply pyproject.toml]: removed

**Comment by steve-barnard** [comment]
All set now 😅

**Pull request merged**